app/src/main/java/com/duckduckgo/app/browser/senseofprotection/SenseOfProtectionExperimentImpl.kt

```diff
@@ -0,0 +1,176 @@
+import com.duckduckgo.feature.toggles.api.MetricsPixel
+import com.duckduckgo.feature.toggles.api.Toggle.State.CohortName
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+private const val EXISTING_USER_DAY_COUNT_THRESHOLD = 28
+
+interface SenseOfProtectionExperiment {
+
+    fun enrolUserInNewExperimentIfEligible(): Boolean
+    fun getTabManagerPixelParams(): Map<String, String>
+    fun firePrivacyDashboardClickedPixelIfInExperiment()
+    fun isUserEnrolledInAVariantAndExperimentEnabled(): Boolean
+    fun isUserEnrolledInVariant2CohortAndExperimentEnabled(): Boolean
+    fun isUserEnrolledInModifiedControlCohortAndExperimentEnabled(): Boolean
+    fun isMemberOfVariant2Cohort(): Boolean
+}
+
+@ContributesBinding(
+    scope = AppScope::class,
+    boundType = SenseOfProtectionExperiment::class,
+)
+@SingleInstanceIn(AppScope::class)
+class SenseOfProtectionExperimentImpl @Inject constructor(
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+    private val userBrowserProperties: UserBrowserProperties,
+    private val senseOfProtectionToggles: SenseOfProtectionToggles,
+    private val senseOfProtectionPixelsPlugin: SenseOfProtectionPixelsPlugin,
+    private val pixel: Pixel,
+) : SenseOfProtectionExperiment {
+
+    init {
+        // enrol users in the existing user experiment if they are not already enrolled in the new user experiment
+        if (userBrowserProperties.daysSinceInstalled() > EXISTING_USER_DAY_COUNT_THRESHOLD) {
+            if (isNotEnrolledInNewUserExperiment()) {
+                isExistingUserExperimentEnabled(cohortName = MODIFIED_CONTROL)
+            }
+        }
+    }
+
+    override fun enrolUserInNewExperimentIfEligible(): Boolean {
+        return if (userBrowserProperties.daysSinceInstalled() <= EXISTING_USER_DAY_COUNT_THRESHOLD) {
+            isNewUserExperimentEnabled(cohortName = MODIFIED_CONTROL)
+        } else {
+            false
+        }
+    }
+
+    override fun isUserEnrolledInVariant2CohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == VARIANT_2.cohortName && isNewUserExperimentEnabled(VARIANT_2) ||
+        getExistingUserExperimentCohortName() == VARIANT_2.cohortName && isExistingUserExperimentEnabled(VARIANT_2)
+
+    override fun isMemberOfVariant2Cohort(): Boolean =
+        senseOfProtectionToggles.senseOfProtectionNewUserExperimentApr25().getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName ||
+            senseOfProtectionToggles.senseOfProtectionExistingUserExperimentApr25().getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName
+
+    override fun getTabManagerPixelParams(): Map<String, String> {
+        return when {
+            isEnrolledInNewUserExperiment() -> {
```

Comment:
We are only checking if the user is enrolled but not if the experiment is enabled.